crates/core/src/transcript/encoding/proof.rs

```diff
@@ -182,3 +182,92 @@ impl From<MerkleError> for EncodingProofError {
         Self::new(ErrorKind::Proof, error)
     }
 }
+
+/// Invalid encoding proof error.
+#[derive(Debug, thiserror::Error)]
+#[error("invalid encoding proof: {0}")]
+pub struct InvalidEncodingProof(&'static str);
+
+mod validation {
+    use super::*;
+
+    /// The maximum allowed height of the Merkle tree of encoding commitments.
+    ///
+    /// The statistical security parameter (SSP) of the encoding commitment
+    /// protocol is calculated as "the number of uniformly random bits in a
+    /// single bit's encoding minus `MAX_HEIGHT`".
+    ///
+    /// For example, a bit encoding used in garbled circuits typically has 127
+    /// uniformly random bits, hence when using it in the encoding
+    /// commitment protocol, the SSP is 117 bits.
+    ///
+    /// DO NOT use bit encodings which have less than 50 uniformly random bits,
+    /// since the SSP < 40 bits is widely considered inadequate.
+    const MAX_HEIGHT: usize = 10;
```

Comment:
is there any reason why 10 is chosen?

Reply:
No specific reason other than we want to achieve SSP=40 and don't want to mandate that encodings are unnecessarily large. It seemes reasonable that there won't be 1024 commitments. Although, now I'm realizing that with our json commit-to-everything strategy 1024 may be insufficient.

Let's tag @sinui0 to double check.